examples/java-matter-controller/java/src/com/matter/controller/commands/common/Argument.java

```diff
@@ -0,0 +1,127 @@
+/*
+ *   Copyright (c) 2022 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+package com.matter.controller.commands.common;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import javax.annotation.Nullable;
+
+public class Argument {
+  private String mName;
+  private ArgumentType mType;
+  private long mMin;
+  private long mMax;
+  private Object mValue;
+  private Optional<String> mDesc = Optional.empty();
```

Comment:
The class and fields here can be final. You will need to ensure defaults are set for each constructor for fields which aren't relevant, but this will ensure fields are initialized on all paths.